miso-dto/src/main/java/uk/ac/bbsrc/tgac/miso/dto/Dtos.java

```diff
@@ -849,28 +852,35 @@ public static LibraryDto asDto(Library from, LibraryAdditionalInfo infoFrom) {
     }
     return dto;
   }
-  
-  public static Library to (LibraryDto from) {
-   Library to = new LibraryImpl();
-   to.setAlias(from.getAlias());
-   to.setName(from.getName());
-   to.setDescription(from.getDescription());
-   to.setInitialConcentration(from.getConcentration());
-   to.setLowQuality(from.getLowQuality());
-   to.setPlatformName(from.getPlatformName());
-   if (from.getTagBarcodeIndex1Id() != null) {
-     HashMap<Integer, TagBarcode> tagBarcodes = new HashMap<>();
-     TagBarcode tb1 = new TagBarcodeImpl();
-     tb1.setId(from.getTagBarcodeIndex1Id());
-     tagBarcodes.put(1, tb1);
-     if (from.getTagBarcodeIndex2Id() != null) {
-       TagBarcode tb2 = new TagBarcodeImpl();
-       tb2.setId(from.getTagBarcodeIndex2Id());
-       tagBarcodes.put(1, tb2);
-     }
-     to.setTagBarcodes(tagBarcodes);
-   }
-   to.setVolume(from.getVolume());
+
+  public static Library to(LibraryDto from) {
+    Library target = new LibraryImpl();
+    return to(from, target);
+  }
+
+  public static Library to(LibraryDto from, Library to) {
```

Comment:
I wrote this. In the other services, we do `update` taking the DTO that modifies the database model. Because the SQLStores aren't designed that way, it made better sense to load the database object, mutate it using the DTO and then save it to the database. That can be reused on a fresh library so that save and update are similar.